### PR TITLE
Updated the multi-node proposal.

### PR DESCRIPTION
**Pull request by amshuman-kr**
**How to categorize this PR?**

**What this PR does / why we need it**:
1. Introduced status.ClusterSize to calculate Ready condition
2. Leading sidecar to cleanup superfluous ETCD members
3. Leading sidecar to defragment itself last after defragmenting others

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
cc @timuthy @ishan16696

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: docs/proposals/multi-node/README.md [hunk 1]
```diff
@@ -581,6 +586,13 @@ The `members` section of the status is intended to be updated by the `etcd-backu
...
+
+This proposal recommends that the `name` of the member (which is the same as the name of the member `Pod`) be used as the unique key to identify a member in the `members` array.
+This can minimise the need to cleanup superfluous entries in the `members` array after the member pods are gone to some extent because the replacement pods for any member will share the same `name` and will overwrite the entry with a [possibly new](#restarting-an-existing-member-of-an-etcd-cluster) member `id`. 
+There is still the possibility of not only [superfluous entries in the `members` array](#13-superfluous-member-entries-in-etcd-status) but also [superfluous `members` in the ETCD cluster](#work-flows-only-on-the-leading-member) for which there is no corresponding pod in the `StatefulSet` anymore both of which need to be [cleaned up](#recommended-action-12) [as appropriate](#work-flows-only-on-the-leading-member).
```
Comment (timuthy) [inline docs/proposals/multi-node/README.md:594]: What do you think about giving an example here:
Cluster is scaled up `3 -> 5`, two new members are added but then constantly crash for OOM reasons. Then cluster is scaled down `5 -> 3` which deletes pods `etcd-main-4`, `etcd-main-5` but `len(status.members) == 5` and `members == 5` in the cluster.

**Review by timuthy — COMMENTED** [review]
Thanks for the update @amshuman-kr. I added one suggestion.

File: docs/proposals/multi-node/README.md (hunk 1 above)
Comment (amshuman-kr) [reply docs/proposals/multi-node/README.md:594]: 👍 Will do

**Comment by amshuman-kr** [comment]
@timuthy I have updated the proposal with an example for superfluous members. Can you PTAL?

**Review by timuthy — APPROVED** [review]
/lgtm
Nice, very appreciate the update!

**Comment by timuthy** [comment]
@ishan16696 is there anything missing from your side? Otherwise we can merge this PR FMPOV.

**Pull request merged**